src/instructlab/model/backends/vllm.py

```diff
@@ -120,32 +164,56 @@ def run_vllm(
         background (bool):            Whether the stdout and stderr vLLM should be sent to /dev/null (True)
                                       or stay in the foreground(False).
     Returns:
-        vllm_process (subprocess.Popen): process of the vllm server
+        tuple: A tuple containing two values:
+            vllm_process (subprocess.Popen): process of the vllm server
+            tmp_files: a list of temporary files necessary to launch the process
+
     """
     vllm_process = None
     vllm_cmd = [sys.executable, "-m", "vllm.entrypoints.openai.api_server"]
+    tmp_files = []
 
-    if "--host" not in vllm_args:
+    if not contains_argument("--host", vllm_args):
         vllm_cmd.extend(["--host", host])
 
-    if "--port" not in vllm_args:
+    if not contains_argument("--port", vllm_args):
         vllm_cmd.extend(["--port", str(port)])
 
-    if "--model" not in vllm_args:
+    if not contains_argument("--model", vllm_args):
         vllm_cmd.extend(["--model", os.fspath(model_path)])
 
+    if not contains_argument("--chat-template", vllm_args):
+        if chat_template == CHAT_TEMPLATE_AUTO:
+            # For auto templates, the build-in in-memory template
+            # needs to be written to a temporary file so that vLLM
+            # can read it
+            template = format_template(model_family, model_path)
+            file = create_tmpfile(template)
+            tmp_files.append(file)
+            vllm_cmd.extend(["--chat-template", file.name])
+        elif chat_template != CHAT_TEMPLATE_TOKENIZER:
+            # In this case the value represents a template file path
+            # Pass it directly to vllm
+            vllm_cmd.extend(["--chat-template", chat_template])
```

Comment:
@n1hility Trying to think of edge cases, Would we want to to make sure the file passed as chat_template exists on the file system? If you pass in a chat template that does not exist to vLLM does it error out and not start up or give a warning of some sort?

 In the cases of `run_vLLM()` being run with background=True, since there is no vLLM output, a user of `ilab`  commands that serve with vLLM under the hood wouldn't see these warning or have vLLM fail in the background.
 
 Maybe we add a check here and some debug logging saying what the chat template file name is if it's set?

Reply:
ah yes this is a good idea for the background case

Reply:
Would you want that to happen for follow up after this PR is merged? I can create an issue, and Seb and I can work on getting that in over the next 24 hours.